File: 347.py
from pstats import Stats
from typing import List
from collections import Counter, defaultdict
import sorting


class Solution:
    # def topKFrequent(self, nums: List[int], k: int) -> List[int]:
    #     counter = {}

    #     for num in nums:
    #         if num not in counter:
    #             counter[num] = 0
    #         counter[num] += 1

    #     a = sorted(counter.items(), key=lambda x: x[1], reverse=True)
    #     res = []
    #     for i in range(k):
    #         res.append(a[i][0])
    #     res = res[::-1]
    #     return res

    # def topKFrequent(self, nums: List[int], k: int) -> List[int]:
    #     sl = sorting.merge(nums)
    #     currNum = {}
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = Counter()
        for n in nums:
            count[n] += 1
        sortedCount = list(count.items())
        sortedCount.sort(key=lambda x: x[1], reverse=True)
        a = list(map(lambda x: x[0], sortedCount))
        return a[:k]


if __name__ == "__main__":
    s = Solution()
    print(Solution.topKFrequent(s, [1, 2, 1, 3, 1, 3], 2))

File: 347.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    if (nums.length === 0) {
        return 0
    }

    const dict = {}

    for (const num of nums) {
        if (dict[num] == null) {
            dict[num] = 0
        }
        dict[num] += 1
    }
    const keyPairs = Object.keys(dict).map((k) => {
        return [k, dict[k]]
    })
    const sortLs = keyPairs.sort((f, s) => {
        return s[1] - f[1]
    })
    
    console.log(sortLs.slice(0,k).map((a) => {return a[0]}));
};

topKFrequent([1,1,1,2,2,3,3,3,3,3,3],2)